src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -202,42 +199,98 @@ public int getDefaultCloudIndexPrefetchBufferSize() {
+                sampleNames.add(sampleName);
+                final String previousPath = sampleNameToVcfUri.put(sampleName, variantPath);
+                if (previousPath != null) {
+                    throw new UserException("Duplicate sample: " + sampleName + ". Sample was found in both "
+                                                    + variantPath + " and " + previousPath + ".");
+                }
             }
+            mergedHeaderLines = VCFUtils.smartMergeHeaders(headers, true);
+            mergedHeaderSequenceDictionary = new VCFHeader(mergedHeaderLines).getSequenceDictionary();
+        } else {
+            // -sampleNameMap was specified
+            sampleNameToVcfUri.putAll(loadSampleNameMapFile(IOUtils.getPath(sampleNameToPathFile)));
+            sampleNames.addAll(sampleNameToVcfUri.keySet());
+
+            final String firstHeaderPath = sampleNameToVcfUri.entrySet().iterator().next().getValue();
+            final VCFHeader header = loadHeaderFromVCFUri(firstHeaderPath);
+            mergedHeaderLines = header.getMetaDataInInputOrder();
+            mergedHeaderSequenceDictionary = header.getSequenceDictionary();
+        }
+    }
 
-            // Import occurs in batches. By closing all readers here we release resources
-            // and preserve memory for later stages where readers are allocated
-            // for samples in a given batch
-            try {
-                reader.close();
-            } catch (IOException e) {
-                throw new GATKException("FeatureReader close() failed for " + sampleName, e);
-            }
+    private VCFHeader loadHeaderFromVCFUri(final String variantPath) {
+        try(final AbstractFeatureReader<VariantContext, LineIterator> reader = getReaderFromVCFUri(variantPath)) {
+            return (VCFHeader) reader.getHeader();
+        } catch (final IOException e) {
+            throw new UserException("Error while reading vcf header from " + variantPath, e);
         }
+    }
 
-        mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
-        mergedVCFHeader = new VCFHeader(mergedHeader);
+    private static void assertGVCFHasOnlyOneSample(String variantPath, VCFHeader header) {
+        // A GVCF file must contain only one sample, throw an exception otherwise
+        final int numberOfSamples = header.getNGenotypeSamples();
+        if (numberOfSamples != 1) {
+            throw new UserException("Input GVCF: " + variantPath + " was expected to contain a single sample but actually contained " + numberOfSamples + ".");
+        }
+    }
 
-        initializeIntervals();
-        super.onStartup();
+    @VisibleForTesting
+    static Map<String, String> loadSampleNameMapFile(final Path sampleToFileMapPath) {
+        try {
+            final Map<String, String> sampleToFilename = Files.lines(sampleToFileMapPath)
+                    .map(line -> {
+                        final String[] split = line.split("\\s");
```

Comment:
Is it useful to require a single whitespace char of any type instead of a single tab? Also might have issues with newlines. Probably safest to just require a tab.